### PR TITLE
docs: add jsdocs to some eslint types

### PR DESCRIPTION
**Pull request by ST-DDT**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - https://github.com/eslint/eslint
  - https://eslint.org/docs/latest/use/configure/

Descriptions partially copied from https://github.com/eslint-types/eslint-define-config with permission.

### CODE REVIEW

**Review by JounQin — APPROVED** [review]

File: types/eslint/index.d.ts [hunk 1]
```diff
@@ -946,7 +1092,21 @@ export namespace Linter {
...
+
+        /**
+         * Set to "script" (default) or "module" if your code is in ECMAScript modules.
+         *
+         * @default 'script'
+         *
+         * @see https://eslint.org/docs/latest/user-guide/configuring/language-options#specifying-parser-options
+         */
         sourceType?: "script" | "module" | undefined;
+        /**
```
Comment (Shinigami92) [inline types/eslint/index.d.ts:1104]: Sometimes there are blank lines above docs, sometimes not
Would you consider to make it consistent?

File: types/eslint/index.d.ts [hunk 2]
```diff
@@ -874,53 +874,199 @@ export class Linter {
...
+
+        /**
+         * Specifying Globals.
+         *
+         * @see [Globals](https://eslint.org/docs/latest/user-guide/configuring/language-options#specifying-globals)
+         */
         globals?: { [name: string]: boolean | "off" | "readonly" | "readable" | "writable" | "writeable" } | undefined;
+
+        /**
+         * Disabling Inline Comments.
```
Comment (Shinigami92) [inline types/eslint/index.d.ts:962]: This title casing looks a bit odd to me 🤔

File: types/eslint/index.d.ts [hunk 3]
```diff
@@ -874,53 +874,199 @@ export class Linter {
...
+         */
         settings?: { [name: string]: any } | undefined;
     }
 
+    /**
+     * The overwrites that apply more differing configuration to specific files or directories.
+     */
     interface ConfigOverride<Rules extends RulesRecord = RulesRecord> extends BaseConfig<Rules> {
         excludedFiles?: string | string[] | undefined;
         files: string | string[];
```
Comment (Shinigami92) [inline types/eslint/index.d.ts:1025]: Are there docs for these two fields you could add?

**Review by Shinigami92 — APPROVED** [review]
You could think about adding your name as an author/maintainer of these docs as the contribution guidelines suggest

All in all, these are just jsdocs added, so nothing that should block a merge 👍

File: types/eslint/index.d.ts (hunk 1 above)
Comment (ST-DDT) [reply types/eslint/index.d.ts:1113]: I tried to make it consistent, but the style of property definitions varies greatly. So I only checked those with jsdocs.

**Review by Shinigami92 — APPROVED** [review]

**Comment by ST-DDT** [comment]
> You could think about adding your name as an author/maintainer of these docs as the contribution guidelines suggest

> The DT maintainers are putting trust in the definition owners to ensure a stable eco-system, please don't add yourself lightly.

I don't think partially adding jsdocs is not enough to qualify for that.
This is probably not the last PR, but I wanted to separate documentation PRs from technical stuff.
I the future, I plan on supporting TS type augmentation for `config.rules` etc and thus providing plugins the capability to support auto completion to the end user for their rules and options.

**Review by andrewbranch — APPROVED** [review]

**Comment by ST-DDT** [comment]
Ready to merge

**Pull request merged**